cmd/collector/app/server/grpc.go

```diff
@@ -42,14 +43,19 @@ type GRPCServerParams struct {
 	MaxReceiveMessageLength int
 	MaxConnectionAge        time.Duration
 	MaxConnectionAgeGrace   time.Duration
+
+	// Set by the server to indicate the actual host:port of the server.
+	HostPortActual string
 }
 
 // StartGRPCServer based on the given parameters
 func StartGRPCServer(params *GRPCServerParams) (*grpc.Server, error) {
 	var server *grpc.Server
 	var grpcOpts []grpc.ServerOption
 
-	grpcOpts = append(grpcOpts, grpc.MaxRecvMsgSize(params.MaxReceiveMessageLength))
+	if params.MaxReceiveMessageLength > 0 {
+		grpcOpts = append(grpcOpts, grpc.MaxRecvMsgSize(params.MaxReceiveMessageLength))
```

Comment:
Adding a 0 value as option actually breaks gRPC message handling/ I guess it wasn't a problem in prod code since the default comes from a CLI flag, but in tests it was annoying having to keep setting this param.